src/oneD/StFlow.cpp

```diff
@@ -344,190 +354,262 @@ void StFlow::updateProperties(size_t jg, double* x, size_t jmin, size_t jmax)
+            for (size_t n = 0; n <= 5; n++) {
+                k_P_H2O += c_H2O[n] * pow(1000 / T(x, j), (double) n);
             }
-            // absorption coefficient for CO2
-            if (m_kRadiating[0] != npos) {
-                double k_P_CO2 = 0;
-                for (size_t n = 0; n <= 5; n++) {
-                    k_P_CO2 += c_CO2[n] * pow(1000 / T(x, j), (double) n);
+            k_P_H2O /= k_P_ref;
+            k_P += m_press * X(x, m_kRadiating[1], j) * k_P_H2O;
+        }
+        // Absorption coefficient for CO2
+        if (m_kRadiating[0] != npos) {
+            double k_P_CO2 = 0;
+            for (size_t n = 0; n <= 5; n++) {
+                k_P_CO2 += c_CO2[n] * pow(1000 / T(x, j), (double) n);
+            }
+            k_P_CO2 /= k_P_ref;
+            k_P += m_press * X(x, m_kRadiating[0], j) * k_P_CO2;
+        }
+
+        // Calculation of the radiative heat loss term
+        double radiative_heat_loss = 0;
+        radiative_heat_loss = 2 * k_P *(2 * StefanBoltz * pow(T(x, j), 4)
+                              - boundary_Rad_left - boundary_Rad_right);
+
+        // set the radiative heat loss vector
+        m_qdotRadiation[j] = radiative_heat_loss;
+    }
+}
+
+void StFlow::evalContinuity(double* x, double* rsd, int* diag,
+                            double rdt, size_t jmin, size_t jmax)
+{
+    // The left boundary has the same form for all cases.
+    if (jmin == 0) { // left boundary
+        rsd[index(c_offset_U,jmin)] = -(rho_u(x,jmin + 1) - rho_u(x,jmin))/m_dz[jmin]
+                                      -(density(jmin + 1)*V(x,jmin + 1)
+                                      + density(jmin)*V(x,jmin));
+    }
+
+    if (jmax == m_points - 1) { // right boundary (same for unstrained/free-flow)
+        rsd[index(c_offset_U, jmax)] = rho_u(x, jmax) - rho_u(x, jmax - 1);
+    }
+
+    // j0 and j1 are constrained to only interior points
+    size_t j0 = std::max<size_t>(jmin, 1);
+    size_t j1 = std::min(jmax, m_points - 2);
+    for (size_t j = j0; j <= j1; j++) {
+        rsd[index(c_offset_U, j)] = rho_u(x, j) - rho_u(x, j - 1);
+        diag[index(c_offset_U, j)] = 0;
+    }
+
+    if (m_usesLambda == false && m_isFree == false) { //unstrained flow
+        return;
+    }
+
+    if (m_isFree == true) { // free flow
+        for (size_t j = j0; j <= j1; j++) {
+            // terms involving V are zero as V=0 by definition
+            if (grid(j) > m_zfixed) {
+                rsd[index(c_offset_U,j)] = -(rho_u(x,j) - rho_u(x,j-1))/m_dz[j-1];
+            } else if (grid(j) == m_zfixed) {
+                if (m_do_energy[j]) {
+                    rsd[index(c_offset_U,j)] = (T(x,j) - m_tfixed);
+                } else {
+                    rsd[index(c_offset_U,j)] = (rho_u(x,j) - m_rho[0]*0.3); // why 0.3?
                 }
-                k_P_CO2 /= k_P_ref;
-                k_P += m_press * X(x, m_kRadiating[0], j) * k_P_CO2;
+            } else if (grid(j) < m_zfixed) {
```

Comment:
```suggestion
            } else { // grid(j) < m_zfixed
```
the check is always true as it's the only case left.